MiniApp/Classes/MiniApp.swift

```diff
@@ -5,36 +5,39 @@ public class MiniApp: NSObject {
     ///         -   Error: Error details if fetching is failed.
-    public class func list(completionHandler: @escaping (Result<[MiniAppInfo], Error>) -> Void) {
-        return RealMiniApp.shared.listMiniApp(completionHandler: completionHandler)
+    public class func list(config: MiniAppSdkConfig? = nil, completionHandler: @escaping (Result<[MiniAppInfo], Error>) -> Void) {
+        return RealMiniApp.shared(with: config).listMiniApp(completionHandler: completionHandler)
     }
 
     /// Fetch the MiniAppInfo information for a given MiniApp id.
     ///
     /// - Parameters:
+    ///     -   config: on optional custom configuration to override default SDK parameters such as base URL, app ID, subsciption key or host app version
     ///     -   miniAppId: the identifier string of the Mini App you want information
     ///     -   completionHandler: A block to be called when MiniAppInfo information is fetched. Completion blocks receives the following parameters
     ///         -   MiniAppInfo: MiniAppInfo information.
     ///         -   Error: Error details if fetching is failed.
-    public class func info(miniAppId: String, completionHandler: @escaping (Result<MiniAppInfo, Error>) -> Void) {
+    public class func info(config: MiniAppSdkConfig? = nil, miniAppId: String, completionHandler: @escaping (Result<MiniAppInfo, Error>) -> Void) {
         if miniAppId.count == 0 {
             return completionHandler(.failure(NSError.invalidAppId()))
         }
-        return RealMiniApp.shared.getMiniApp(miniAppId: miniAppId, completionHandler: completionHandler)
+        return RealMiniApp.shared(with: config).getMiniApp(miniAppId: miniAppId, completionHandler: completionHandler)
     }
 
     /// Create a Mini App for the given mini app info object, Mini app will be downloaded and cached in local.
     ///
     /// - Parameters:
+    ///   - config: on optional custom configuration to override default SDK parameters such as base URL, app ID, subsciption key or host app version
     ///   - appInfo: Mini App info object
     ///   - completionBlock: A block to be called on successful creation of [MiniAppView] or throws errors if any. Completion blocks receives the following parameters
     ///         -   MiniAppDisplayProtocol: Protocol that helps the hosting application to communicate with the displayer module of the mini app. More like an interface for host app
     ///                         to interact with View component of mini app.
     ///         -   Error: Error details if Mini App View creating is failed
-    public class func create(appInfo: MiniAppInfo, completionHandler: @escaping (Result<MiniAppDisplayProtocol, Error>) -> Void) {
-        return RealMiniApp.shared.createMiniApp(appInfo: appInfo, completionHandler: completionHandler)
+    public class func create(config: MiniAppSdkConfig? = nil, appInfo: MiniAppInfo, completionHandler: @escaping (Result<MiniAppDisplayProtocol, Error>) -> Void) {
```

Comment:
I was looking at the Android PR before this. In iOS, I see that using `MiniApp.swift` the functions are being accessed statically on the app side. On the Android side, even though a single instance is being maintained for this, it is still accessed via the object. @minh-rakuten leveraged this and has handled `config` differently. It looks like a nice idea, what do you think @Climbatize ?
It has been working fine as the public APIs were more or less the same for both the platforms. Even after this, a small deviation shouldn't impact as such 💭

Reply:
@ts-leojoseph-r it is only a single instance with ability to update the configuration.

Reply:
I moved the singleton from RealMiniApp to MiniApp to match Android